test/alt.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
```

Comment:
Are you suggesting defining a type which provides *only* the methods and/or TypeRep functions necessary to satisfy the requirements of the type class?